### PR TITLE
feat: variable binding

### LINKED ISSUE #1220: [Content v2] Use frontmatter settings inside md
### Is your feature request related to a problem? Please describe
Netlify CMS provides you with the option to save frontmatter variables without touching the underlying .md file. I would love to use netlify CMS on top of content v2, but it seems that I can't reuse frontmatter vars inside the .md itself.
### Describe the solution you'd like
```md
---
some_var_inserted_by_netlify_cms: This is a test
---

# Check this title from netlify cms
## ${some_var_inserted_by_netlify_cms}
```

### Describe alternatives you've considered
I've tried installing the `remark-mdx-frontmatter` plugin but didn't work.

### PR DESCRIPTION
**Pull request by farnabaz**
### 🔗 Linked issue
resolves #1220
### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Introducing new syntax for interpolating text in markdown.
Screen Shot 2022-06-15 at 3 04 45 PM

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/runtime/components/MarkdownRenderer.ts [hunk 1]
```diff
@@ -106,58 +106,86 @@ export default defineComponent({
...
   )
 }
 
+function renderBinding (node: MarkdownNode, h: CreateElement, documentMeta: ParsedContentMeta, parentScope: any = {}): ContentVNode {
+  const data = {
+    ...parentScope,
+    $route: () => useRoute(),
+    $document: documentMeta,
+    $doc: documentMeta
+  }
```
Comment (farnabaz) [inline src/runtime/components/MarkdownRenderer.ts:144]: We should define what should be available in interpolation.

Comment (Atinux) [reply src/runtime/components/MarkdownRenderer.ts:144]: I think this is a good start. Also we will need to see how we can support this in the editor 🤔

**Review by Atinux — APPROVED** [review]

**Comment by ManasMadrecha** [comment]
Is the documentation updated? Because had to search a lot into Issues to get to this image to figure out that to send v-bind props, we don't need `$doc.`

Thanks for this wonderful update BTW. 😄

**Comment by Atinux** [comment]
It has not been updated, though, a PR is more than welcome if you want to help :green_heart:

**Pull request merged**